src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -975,22 +877,198 @@ impl MooncakeTable {
+        tokio::fs::remove_dir_all(&self.metadata.path).await?;
+        Ok(())
+    }
+
+    /// Drop an iceberg table.
+    pub(crate) async fn drop_iceberg_table(&mut self) -> Result<()> {
+        assert!(self.iceberg_table_manager.is_some());
+        self.iceberg_table_manager
+            .as_mut()
+            .unwrap()
+            .drop_table()
+            .await?;
+        Ok(())
+    }
+
+    /// Uses the latest wal metadata and iceberg LSN from the latest iceberg snapshot
+    /// to determine the files to truncate.
+    ///
+    /// # Arguments
+    ///
+    /// * uuid: WAL persistence event unique id.
+    #[must_use]
+    pub(crate) fn do_wal_persistence_update(&mut self, uuid: uuid::Uuid) -> bool {
+        let latest_iceberg_snapshot_lsn = self.get_iceberg_snapshot_lsn();
+
+        let wal_persistence_update_result = self
+            .wal_manager
+            .prepare_persistent_update(latest_iceberg_snapshot_lsn);
+
+        if wal_persistence_update_result.should_do_persistence() {
+            let event_sender_clone = self.table_notify.as_ref().unwrap().clone();
+            let file_system_accessor = self.wal_manager.get_file_system_accessor();
+            tokio::spawn(async move {
+                WalManager::wal_persist_truncate_async(
+                    uuid,
+                    wal_persistence_update_result,
+                    file_system_accessor,
+                    event_sender_clone,
+                )
+                .await;
+            });
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Handles the result of a persist and truncate operation.
+    /// Returns the highest LSN that has been persisted into WAL.
+    pub(crate) fn handle_completed_wal_persistence_update(
+        &mut self,
+        result: &WalPersistenceUpdateResult,
+    ) -> Option<u64> {
+        self.wal_manager
+            .handle_complete_wal_persistence_update(result)
+    }
+
+    /// Drop the WAL files. Note that at the moment this drops WAL files in the mooncake table's local filesystem
+    /// and so behaves the same as MooncakeTable.drop_table(),
+    /// but in the future this is needed to support object storage
+    pub(crate) async fn drop_wal(&mut self) -> Result<()> {
+        self.wal_manager.drop_wal().await
+    }
+
+    pub fn push_wal_event(&mut self, event: &TableEvent) {
+        self.wal_manager.push(event);
+    }
+
+    /// Shutdown the current table, which unpins all referenced data files in the global data file.
+    pub async fn shutdown(&mut self) -> Result<()> {
+        let evicted_files_to_delete = {
+            let mut guard = self.snapshot.write().await;
+            guard.unreference_and_delete_all_cache_handles().await
+        };
+
+        for cur_file in evicted_files_to_delete.into_iter() {
+            tokio::fs::remove_file(cur_file).await?;
+        }
+
+        Ok(())
+    }
+
+    /// =======================
```

Comment:
reorder the table events, so all below functions should have its own event serialized and recorded for replay